### PR TITLE
Operator control sensitivity

### LINKED ISSUE #12: Prototyping: Operator Control Sensitivity
complaint is that Mergonaut controls were too sensitive 
what can we do to adjust this? (square inputs etc)
do different controller options help?
can we make this user adjustable?
will need feedback from a 2019 driver.

### PR DESCRIPTION
**Pull request by AbhiJ2706**
Added sensitive operator control. This allows the operator to steer the robot at a much smaller max speed for more precise aiming when shooting.

Issue #12

### CODE REVIEW

**Comment by AbhiJ2706** [comment]
Based on an outdated master. I will fix this before merging.

**Comment by AbhiJ2706** [comment]
Ready for review.

File: src/main/java/frc/robot/commands/SensitiveDriverControl.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        
+    }
+
+    /**
+     * Drives with sensitive control
+     */
+    @Override
+    public void execute() {
+        this.driveBase.sensitiveSteering = true;
+        this.driveBase.tankDrive(j.getRawAxis(4), j.getRawAxis(4), false);
```
Comment (georgetzavelas) [inline src/main/java/frc/robot/commands/SensitiveDriverControl.java]: 4 is a magic number. Can you assign it to the constant?

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -70,10 +77,14 @@ private void configureButtonBindings() {
...
          * Select drive mode for robot
          */
-        
+      
+        // Instantiate the intake command and bind it
         driveCommand = new ArcadeDriveWithJoystick(driverStick, Config.LEFT_CONTROL_STICK_Y, Config.INVERT_FIRST_AXIS, Config.RIGHT_CONTROL_STICK_X, Config.INVERT_SECOND_AXIS);
         DriveBase.getInstance().setDefaultCommand(driveCommand);
 
+        sensitiveDriverControlCommand = new SensitiveDriverControl(driverStick);
+        new JoystickButton(driverStick, XboxController.Button.kBumperLeft.value).whenHeld(sensitiveDriverControlCommand);
```
Comment (georgetzavelas) [inline src/main/java/frc/robot/RobotContainer.java]: What are your thoughts on moving this into the `SensitiveDriverControl` class?

File: src/main/java/frc/robot/commands/SensitiveDriverControl.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+public class SensitiveDriverControl extends CommandBase {
+    
+    //get the drivebase
+    private DriveBase driveBase;
+    private Joystick j;
+
+    public SensitiveDriverControl(Joystick j) {
+        //set the drivebase
+        addRequirements(DriveBase.getInstance());
+        this.driveBase = DriveBase.getInstance();
```
Comment (georgetzavelas) [inline src/main/java/frc/robot/commands/SensitiveDriverControl.java:29]: nitpick: I would put this line first and then assign it to the addRequirements, e.g.
```
this.driveBase = DriveBase.getInstance();
addRequirements(this.driveBase);
```

File: src/main/java/frc/robot/commands/SensitiveDriverControl.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import frc.robot.subsystems.*;
+
+/**
+ * Triggers a modifier which limits the speed the robot can turn and only allows it to turn in place
+ */
+public class SensitiveDriverControl extends CommandBase {
+    
+    //get the drivebase
+    private DriveBase driveBase;
+    private Joystick j;
```
Comment (georgetzavelas) [inline src/main/java/frc/robot/commands/SensitiveDriverControl.java]: Can you give this variable a more meaningful name?

**Review by georgetzavelas — COMMENTED** [review]
Nice work Abhi! Some comments for your consideration

**Review by georgetzavelas — APPROVED** [review]

**Review by Lockscout — APPROVED** [review]
Looks pretty good. Ill see how much it screws with tank drive later

**Pull request merged**